File: user-service/src/main/java/com/oognuyh/userservice/config/KafkaTopicConfig.java
package com.oognuyh.userservice.config;

import org.apache.kafka.clients.admin.NewTopic;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.kafka.config.TopicBuilder;

@Configuration
public class KafkaTopicConfig {
    @Value("${spring.kafka.template.user-changed-topic}")
    private String USER_CHANGED_TOPIC;

    @Bean
    public NewTopic userChangedTopic() {
        return TopicBuilder
            .name(USER_CHANGED_TOPIC)
            .build();
    }
}

File: notification-service/src/main/kotlin/com/oognuyh/notificationservice/config/KafkaTopicConfig.kt
package com.oognuyh.notificationservice.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.kafka.config.TopicBuilder

@Configuration
class KafkaTopicConfig(
    @Value("\${spring.kafka.template.notification-topic}") private val NOTIFICATION_TOPIC: String
) {

    @Bean
    fun notificationTopic() = TopicBuilder
        .name(NOTIFICATION_TOPIC)
        .build()
}